pkg/speaker/bgp/policy.go

```diff
@@ -0,0 +1,116 @@
+package bgp
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/openelb/openelb/pkg/constant"
+	"github.com/openelb/openelb/pkg/speaker/bgp/config"
+	"github.com/openelb/openelb/pkg/speaker/bgp/table"
+	api "github.com/osrg/gobgp/api"
+	"github.com/osrg/gobgp/pkg/server"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (b *Bgp) UpdatePolicy(cm *corev1.ConfigMap) error {
+	policyConf, ok := cm.Data[constant.OpenELBBgpName]
+	if !ok {
+		b.log.Info("error in %s configmap, %s missing", constant.OpenELBBgpName)
+		return nil
+	}
+	path, err := writeToTempFile(policyConf)
+	defer os.RemoveAll(path)
+	if err != nil {
+		return err
+	}
+	newConfig, err := config.ReadConfigfile(path, "toml")
+	if err != nil {
+		return err
+	}
+	p := config.ConfigSetToRoutingPolicy(newConfig)
+	rp, err := table.NewAPIRoutingPolicyFromConfigStruct(p)
+	if err != nil {
+		b.log.Error(err, "failed to update policy config")
+		return err
+	}
+	err = b.bgpServer.SetPolicies(context.Background(), &api.SetPoliciesRequest{
+		DefinedSets: rp.DefinedSets,
+		Policies:    rp.Policies,
+	})
+	if err != nil {
+		b.log.Info("successfully updated policy config")
```

Comment:
Why does `err` not equal `nil`, but still output the `successfully updated policy config` log?

Reply:
When an error occurs, consider whether the error log level should be used